### DISCUSSION TITLE
NextJS is not rewriting url() to include file hashes

### DISCUSSION THREAD

**Discussion by jleclanche** [vercel/next.js#43373] [Ideas]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

15

**What browser are you using?**

-

**What operating system are you using?**

Linux

**How are you deploying your application?**

next start

**Describe the Bug**

When using url() in css files and running `yarn build`, the css `url()` inclusions are not rewritten to include the file hash, which makes the URLs point to the version in /public instead of the version in .next/static.

This incidentally leads to two things:
- In a normal deployment, the same file may live on two different URLs. For example if you have an `icon.png` that is included as an `<img>` in the code, and a `background-image` in css, it will result in two separate queries at two separate paths for it.
- Unworkable behaviour when using an assetPrefix: Syncing .next/static/ to a cdn is not enough to serve everything under that assetPrefix.

**Expected Behavior**

The url() contents should be rewritten to include the file hash as part of the filename.

**To Reproduce**

Include a file in a css/scss file via, eg, the following:

```css
.test {
    background: url("/images/test.png");
}
```

Run a build with `yarn build`. The resulting css includes `test.png`, instead of `test.[hash].png` (which is the version present in .next/static/)

Tested with: scss modules & node-sass.

**Comment by jankaifer** [comment]
This is still true in `next@13`.
Repro: https://github.com/JanKaifer/next-repro-21474

**Comment by jankaifer** [comment]
At the same time, this is not a bug because `<img src="/images/test.png" />` behaves exactly the same way. Image optimizations are specific to `next/image`.

I'll convert this into a discussion - that's how we track feature requests.
